### PR TITLE
Fixing some issues related to models-ngspice for GF180MCU

### PR DESCRIPTION
**Pull request by FaragElsayed2**
## Fixing some issues related to models-ngspice for GF180MCU

### CODE REVIEW

**Comment by FaragElsayed2** [comment]
@atorkmabrains Ready to review

File: models/ngspice/testing/regression/diode/models_regression.py [hunk 1]
```diff
@@ -594,8 +593,10 @@ def main():
                 )
                 # fill nan values with 0
                 result_data["error"] = result_data["error"].fillna(0)
+
                 # get rms error
-                result_data["rms_error"] = np.sqrt(np.mean(result_data["error"] ** 2))
+                result_data["rms_error"] = np.sqrt(np.mean((result_data["error"] - result_data["error"].mean()) ** 2))
```
Comment (atorkmabrains) [inline models/ngspice/testing/regression/diode/models_regression.py:598]: @FaragElsayed2 This equation is incorrect.

File: models/ngspice/testing/regression/mos_iv_vbs/models_regression.py [hunk 2]
```diff
@@ -561,11 +560,11 @@ def error_cal(
         result_data[f"measured_vbs={mos_vbs[4]}"] = result_data[f"measured_vbs={mos_vbs[4]}"].clip(
             lower=lowest_curr
         )
-        result_data[f"simulated_vbs={mos_vbs[0]}"] = result_data[f"simulated_vbs={mos_vbs[0]}"].clip(lower=lowest_curr)
-        result_data[f"simulated_vbs={mos_vbs[1]}"] = result_data[f"simulated_vbs={mos_vbs[1]}"].clip(lower=lowest_curr)
-        result_data[f"simulated_vbs={mos_vbs[2]}"] = result_data[f"simulated_vbs={mos_vbs[2]}"].clip(lower=lowest_curr)
-        result_data[f"simulated_vbs={mos_vbs[3]}"] = result_data[f"simulated_vbs={mos_vbs[3]}"].clip(lower=lowest_curr)
-        result_data[f"simulated_vbs={mos_vbs[4]}"] = result_data[f"simulated_vbs={mos_vbs[4]}"].clip(lower=lowest_curr)
+        result_data[f"simulated_vbs={mos_vbs[0]}"] = result_data[f"simulated_vbs={mos_vbs[0]}"].where(result_data[f"measured_vbs={mos_vbs[0]}"] != lowest_curr, lowest_curr)
```
Comment (atorkmabrains) [inline models/ngspice/testing/regression/mos_iv_vbs/models_regression.py:563]: @FaragElsayed2 Why you changed clip?

File: models/ngspice/testing/regression/mos_iv_vbs/models_regression.py [hunk 3]
```diff
@@ -592,18 +591,27 @@ def error_cal(
             * 100.0
             / (result_data[f"measured_vbs={mos_vbs[4]}"])
         )
-        result_data["error"] = (
+
+        # RMS error for all Vbs sweeps
```
Comment (atorkmabrains) [inline models/ngspice/testing/regression/mos_iv_vbs/models_regression.py:595]: @FaragElsayed2 Please change the equation or we need to discsuss.

**Review by atorkmabrains — COMMENTED** [review]
@FaragElsayed2 I have left some comments. Please fix.

**Comment by atorkmabrains** [comment]
@FaragElsayed2 I have added some extra comments.

**Review by atorkmabrains — COMMENTED** [review]
@FaragElsayed2 More comments

File: models/ngspice/testing/180MCU_SPICE_DATA_clean/nfet_03v3_sweeps.csv [hunk 4]
```diff
@@ -0,0 +1,226 @@
+,W (um),L (um),corner,temp,const_var,const_var_val,sweeps,out_col
```
[the remark this answers is not in the document]
Comment (FaragElsayed2) [reply models/ngspice/testing/180MCU_SPICE_DATA_clean/nfet_03v3_sweeps.csv]: Done

File: models/ngspice/testing/180MCU_SPICE_DATA_clean/convert_foundry_csv.py [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+        "id",
+        "rds",
+        "const_var",
+        "const_var_val",
+        "sweeps",
+        "out_col",
+    ]
+    df_rds_vds_vgs = df_rds_vds_vgs.reindex(columns=cols_order_group)
+
+    return df_rds_vds_vgs
```
[the remark this answers is not in the document]
Comment (FaragElsayed2) [reply models/ngspice/testing/180MCU_SPICE_DATA_clean/convert_foundry_csv.py:236]: Done

File: models/ngspice/testing/180MCU_SPICE_DATA_clean/convert_foundry_csv.py [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+            "temp",
+            "const_var",
+            "const_var_val",
+            "sweeps",
+            "out_col",
+        ]
+    ].copy()
+
+    sweeps_df.drop_duplicates(inplace=True)
+    sweeps_df.to_csv(f"{dev_name}_sweeps.csv")
```
[the remark this answers is not in the document]
Comment (FaragElsayed2) [reply models/ngspice/testing/180MCU_SPICE_DATA_clean/convert_foundry_csv.py]: Done

File: models/ngspice/testing/regression/mos_iv_vbs/models_regression.py (hunk 3 above)
Comment (FaragElsayed2) [reply models/ngspice/testing/regression/mos_iv_vbs/models_regression.py:595]: Done in new regression

File: models/ngspice/testing/regression/mos_iv_vbs/models_regression.py (hunk 2 above)
Comment (FaragElsayed2) [reply models/ngspice/testing/regression/mos_iv_vbs/models_regression.py:563]: Done in new regression

File: models/ngspice/testing/regression/diode/models_regression.py (hunk 1 above)
Comment (FaragElsayed2) [reply models/ngspice/testing/regression/diode/models_regression.py:598]: We agreed to use new method for error calculation.

**Comment by FaragElsayed2** [comment]
@atorkmabrains Could you please merge this ?

I will continue on another PR to be able to track changes.

**Review by atorkmabrains — APPROVED** [review]
Thanks @FaragElsayed2

**Pull request merged**